config/jupyterhub_config.py

```diff
@@ -4,24 +4,40 @@
 https://jupyterhub.readthedocs.io/en/latest/tutorial/getting-started/config-basics.html
 https://jupyterhub.readthedocs.io/en/stable/reference/config-reference.html
 """
+
 import os
 
+import nativeauthenticator
+
 from jupyterhub_config.custom_spawner import VirtualEnvSpawner
 
 c = get_config()
 
-# Set the authenticator class
-# TODO: Change the authenticator class to a secure one (e.g. GitHubOAuthenticator)
-c.JupyterHub.authenticator_class = 'jupyterhub.auth.DummyAuthenticator'
-c.Authenticator.allowed_users = {'spark_user', 'test_user1', 'test_user2'}
-c.DummyAuthenticator.password = os.environ['JUPYTERHUB_ADMIN_PASSWORD']
+# Set the authenticator class to nativeauthenticator
+# ref: https://native-authenticator.readthedocs.io/en/latest/quickstart.html
+c.JupyterHub.authenticator_class = 'native'
+c.JupyterHub.template_paths = [f"{os.path.dirname(nativeauthenticator.__file__)}/templates/"]
+c.NativeAuthenticator.open_signup = True
+c.NativeAuthenticator.check_common_password = True
+c.NativeAuthenticator.minimum_password_length = 6
 
+# Set up the admin user
 c.Authenticator.admin_users = {'spark_user'}
+# TODO set admin user password to os.environ['JUPYTERHUB_ADMIN_PASSWORD'] automatically - currently spark_user is created manually with the signup page
+c.Authenticator.allow_all = True
+
+c.JupyterHub.cookie_secret_file = f"{os.environ['JUPYTERHUB_USER_HOME']}/jupyterhub_cookie_secret"
```

Comment:
What is this for?

Reply:
keep cookies I guess.

Reply:
Should it be in the user home dir if it's holding secrets, then?

Reply:
It's a global file. Not created individually for each user.

Reply:
That's why I'm asking why it should be in the home dir at all

Reply:
It's just a convenient to put all jupyterhub related files.

Reply:
I would definitely not put password / secret information in the user directory hierarchy

Reply:
👍

Reply:
:+1: